From edba7d5b88727f14969444628bd8da420b8b5e71 Mon Sep 17 00:00:00 2001
From: shyjumon <shyjumon.n@intel.com>
Date: Tue, 28 May 2019 07:09:24 +0530
Subject: [PATCH] HACK: mmc: sdhci-pci: Disable DCMD for CNP

DCMD will be fixed in PCH C0 refer HSD 1604306740
ported from 5.x kernel of Hunter.

Tracked-On: None
Signed-off-by: shyjumon <shyjumon.n@intel.com>
Signed-off-by: Adrian Hunter <adrian.hunter@intel.com>
---
 drivers/mmc/host/sdhci-pci-core.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-pci-core.c b/drivers/mmc/host/sdhci-pci-core.c
index a8d5160..6a86d82 100644
--- a/drivers/mmc/host/sdhci-pci-core.c
+++ b/drivers/mmc/host/sdhci-pci-core.c
@@ -745,7 +745,8 @@ static int glk_emmc_probe_slot(struct sdhci_pci_slot *slot)
 		slot->host->mmc->caps2 |= MMC_CAP2_HS400_ES,
 		slot->host->mmc_host_ops.hs400_enhanced_strobe =
 						intel_hs400_enhanced_strobe;
-		slot->host->mmc->caps2 |= MMC_CAP2_CQE_DCMD;
+		if (slot->chip->pdev->device != PCI_DEVICE_ID_INTEL_CNP_EMMC)
+			slot->host->mmc->caps2 |= MMC_CAP2_CQE_DCMD;
 	}
 
 	return ret;
-- 
2.7.4

